File: example - copia/coach.php
<?php

require_once 'person.php';
require_once 'health.php';
require_once 'pension.php';
require_once 'arl.php';

class coach extends person {
    private $valueDays;
    private $numDays;
    private $totalClients;
    private $healthDeduction;
    private $pensionDeduction;
    private $arlDeduction;

    public function __construct(int $identification, string $job, int $transport, int $valueDays, int $numDays) {
        parent::__construct($identification, $job, $transport);
        $this->valueDays = $valueDays;
        $this->numDays = $numDays;
        $this->totalClients = 0;
        $this->healthDeduction = new Health();
        $this->pensionDeduction = new Pension();
        $this->arlDeduction = new Arl();
    }

    public function getValueDays(): int {
        return $this->valueDays;
    }

    public function setValueDays(int $valueDays): void {
        $this->valueDays = $valueDays;
    }

    public function getNumDays(): int {
        return $this->numDays;
    }

    public function setNumDays(int $numDays): void {
        $this->numDays = $numDays;
    }

    public function getTotalClients(): int {
        return $this->totalClients;
    }

    public function setTotalClients(int $totalClients): void {
        $this->totalClients = $totalClients;
    }

    public function addClient(): void {
        $this->totalClients++;
    }

    public function calculateBaseSalary(): float {
        return ($this->valueDays * $this->numDays) + (100 * $this->totalClients);
    }

    public function calculateDeductions(float $salary): float {
        return $this->healthDeduction->discounts($salary) +
               $this->pensionDeduction->discounts($salary) +
               $this->arlDeduction->discounts($salary);
    }
}

?>

File: example - copia/java/coach.java
public class coach extends person {
    private int valueDays;
    private int numDays;
    private int totalClients;
    private final health healthDeduction;
    private final pension pensionDeduction;
    private final arl arlDeduction;

    public coach(int identification, String job, int transport, int valueDays, int numDays) {
        super(identification, job, transport);
        this.valueDays = valueDays;
        this.numDays = numDays;
        this.totalClients = 0;
        this.healthDeduction = new health();
        this.pensionDeduction = new pension();
        this.arlDeduction = new arl();
    }

    public int getValueDays() {
        return valueDays;
    }

    public void setValueDays(int valueDays) {
        this.valueDays = valueDays;
    }

    public int getNumDays() {
        return numDays;
    }

    public void setNumDays(int numDays) {
        this.numDays = numDays;
    }

    public int getTotalClients() {
        return totalClients;
    }

    public void setTotalClients(int totalClients) {
        this.totalClients = totalClients;
    }

    public void addClient() {
        this.totalClients++;
    }

    public double calculateBaseSalary() {
        return (this.valueDays * this.numDays) + (100 * this.totalClients);
    }

    public double calculateDeductions(double salary) {
        return this.healthDeduction.discounts(salary) +
               this.pensionDeduction.discounts(salary) +
               this.arlDeduction.discounts(salary);
    }
}